File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findall();
    }

    @RequestMapping("/findPage")
    public PageResult pagefind(int page, int rows) {

        PageResult pageResult = brandService.findPage(page, rows);
        return pageResult;
    }

    @RequestMapping("/add")
    public Result addbrand(@RequestBody TbBrand brand) {

        System.out.println();
        try {

            int addbrand = brandService.addbrand(brand);
            Result result = new Result(true, "增加成功");
            return result;
        } catch (Exception e) {

            Result result = new Result(false, "增加失败");
            e.printStackTrace();
            return result;
        }

    }

    @RequestMapping("/findOne")
    public TbBrand findOne(long id) {
        return brandService.findOne(id);

    }

    @RequestMapping("/update")
    public Result updatebrand(@RequestBody TbBrand tbBrand) {
        try {
            int i = brandService.update(tbBrand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    @RequestMapping("/delete")
    public Result deleteBrand(long[] ids) {

        try {
            brandService.delete(ids);
            return new Result(true, "清除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "清除成功");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand, int page, int rows) {

        PageResult seach = brandService.seach(brand, page, rows);
        return seach;
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope,$controller, brandService) {
$controller('baseController',{$scope:$scope})

    //读取列表数据绑定到表单中
    //不分页查询
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        );
    }

    //分页
    $scope.findPage = function (page, rows) {
        brandService.findPage(page, rows).success(
            function (response) {
                $scope.list = response.rows;//显示当前页数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    }

    //根据id查询
    $scope.findOne = function (id) {
        alert(id);
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        );
    }


    //添加和编辑
    $scope.save = function () {
        brandService.save($scope.entity)
            .success(
                function (response) {
                    if (response.success) {
                        //重新查询
                        $scope.reloadList();//重新加载
                    } else {
                        alert(response.message);
                    }
                })
    }

    //删除
    $scope.delete = function () {
        alert($scope.selectIds);
        brandService.delete($scope.selectIds).success(function (data) {
            if (data.success) {
                $scope.reloadList();//重新加载
            }

        })
    }

    //条件查询
    $scope.search = function (page, rows) {

        brandService.search(page, rows, $scope.searchEntity).success(function (response) {
            $scope.paginationConf.totalItems = response.total;//总记录数
            $scope.list = response.rows;//给列表变量赋值

        })
    }

});